assembly/src/parsers/ast/nodes.rs

```diff
@@ -0,0 +1,253 @@
+    U32CheckedDivModImm(u32),
+    U32UncheckedDivMod,
+    U32UncheckedDivModImm(u32),
+    U32CheckedAnd,
+    U32CheckedOr,
+    U32CheckedXor,
+    U32CheckedNot,
+    U32CheckedShr,
+    U32CheckedShrImm(u8),
+    U32UncheckedShr,
+    U32UncheckedShrImm(u8),
+    U32CheckedShl,
+    U32CheckedShlImm(u8),
+    U32UncheckedShl,
+    U32UncheckedShlImm(u8),
+    U32CheckedRotr,
+    U32CheckedRotrImm(u8),
+    U32UncheckedRotr,
+    U32UncheckedRotrImm(u8),
+    U32CheckedRotl,
+    U32CheckedRotlImm(u8),
+    U32UncheckedRotl,
+    U32UncheckedRotlImm(u8),
+    U32CheckedEq,
+    U32CheckedEqImm(u32),
+    U32CheckedNeq,
+    U32CheckedNeqImm(u32),
+    U32CheckedLt,
+    U32UncheckedLt,
+    U32CheckedLte,
+    U32UncheckedLte,
+    U32CheckedGt,
+    U32UncheckedGt,
+    U32CheckedGte,
+    U32UncheckedGte,
+    U32CheckedMin,
+    U32UncheckedMin,
+    U32CheckedMax,
+    U32UncheckedMax,
+
+    // ----- stack manipulation ---------------------------------------------------------------
+    Drop,
+    DropW,
+    PadW,
+    Dup0,
+    Dup1,
+    Dup2,
+    Dup3,
+    Dup4,
+    Dup5,
+    Dup6,
+    Dup7,
+    Dup8,
+    Dup9,
+    Dup10,
+    Dup11,
+    Dup12,
+    Dup13,
+    Dup14,
+    Dup15,
+    DupW0,
+    DupW1,
+    DupW2,
+    DupW3,
+    Swap,
+    Swap2,
+    Swap3,
+    Swap4,
+    Swap5,
+    Swap6,
+    Swap7,
+    Swap8,
+    Swap9,
+    Swap10,
+    Swap11,
+    Swap12,
+    Swap13,
+    Swap14,
+    Swap15,
+    SwapW,
+    SwapW2,
+    SwapW3,
+    SwapDW,
+    MovUp2,
+    MovUp3,
+    MovUp4,
+    MovUp5,
+    MovUp6,
+    MovUp7,
+    MovUp8,
+    MovUp9,
+    MovUp10,
+    MovUp11,
+    MovUp12,
+    MovUp13,
+    MovUp14,
+    MovUp15,
+    MovUpW2,
+    MovUpW3,
+    MovDn2,
+    MovDn3,
+    MovDn4,
+    MovDn5,
+    MovDn6,
+    MovDn7,
+    MovDn8,
+    MovDn9,
+    MovDn10,
+    MovDn11,
+    MovDn12,
+    MovDn13,
+    MovDn14,
+    MovDn15,
+    MovDnW2,
+    MovDnW3,
+    CSwap,
+    CSwapW,
+    CDrop,
+    CDropW,
+
+    // ----- input / output operations --------------------------------------------------------
+    Adv(Felt),
+    Locaddr(Felt),
+    Sdepth,
+    MemLoad,
+    MemLoadImm(Felt),
+    MemLoadW,
+    MemLoadWImm(Felt),
+    LocLoad(Felt),
+    LocLoadW(Felt),
+    MemStore,
+    MemStoreImm(Felt),
+    LocStore(Felt),
+    MemStoreW,
+    MemStoreWImm(Felt),
+    LocStoreW(Felt),
+    LoadWAdv,
+
+    PushConstants(Vec<Felt>),
+    AdvU64Div,
+    AdvPush(Felt),
+    AdvLoadW(Felt),
+
+    // ----- cryptographic operations ---------------------------------------------------------
+    RPHash,
+    RPPerm,
+    MTreeGet,
+    MTreeSet,
+    MTreeCWM,
+
+    // ----- exec / call ----------------------------------------------------------------------
+    ExecLocal(u32),
+    ExecImported(String),
+    CallLocal(u32),
+    CallImported(String),
+}
+
+// A node in a AST that can represent a block, instruction or a control flow.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum Node {
```

Comment:
nit: I'd move `Node` above `Instruction` in this file.

Reply:
fixed